ITISFoundation/osparc-simcore
# 🐛  Fix: file uploads due to bad path encoding

## What do these changes do?


This update resolves an issue with file uploads via drag-and-drop when using the file picker.

### Issue Explanation

The problem stemmed from multiple encodings of the `/` symbol in URLs when converting from web API URLs to storage API URLs. This bug reappeared after an upgrade to the [`yarl`](https://yarl.aio-libs.org/en/stable/api/) library, which introduced changes to its URL handling behavior. For more information, refer to the `test_yarl_new_url_generation`.

## Related Issue(s)

- @sanderegg reported this behaviour in master deploy

## How to test

1. Login in osparc and enable developer mode (F2)
2. Open an existing project.
3. Drag and drop a file onto the workbench.
4. Verify that the file upload works as expected. All responses in `storage` returned w/o errors

**Note:** This pull request requires a follow-up to enhance the `test_openapi_regression_test`. The current test is not fully effective because it mocks the URL converters and the responses from the storage API, which limits its ability to catch what the test was supposed to test :-) 

## DevOps Checklist

None
